src/main/scala/viper/gobra/reporting/StatsCollector.scala

```diff
@@ -0,0 +1,376 @@
+
+    // Check whether the program containing this node has the builtin tag
+    val isBuiltIn = nodeTypeInfo.program(p).isBuiltin
+    // Check if a node comes from a import, used to declare it non abstract, since imports are per default always
+    // abstract and we don't want to generate unnecessary warnings
+    val isImported = !typeInfo.tree.originalRoot.eq(nodeTypeInfo.tree.originalRoot)
+
+    p match {
+      case p: PFunctionDecl =>
+        GobraMemberInfo(pkgDir, pkgName, p.id.name, formatArgs(p.args), FunctionDeclaration, isTrusted = p.spec.isTrusted, p.body.isEmpty && !isImported, isImported, isBuiltIn)
+      case p: PMethodDecl =>
+        GobraMemberInfo(pkgDir, pkgName, p.receiver.typ.formattedShort + "." + p.id.name, formatArgs(p.args), MethodDeclaration, p.spec.isTrusted, p.body.isEmpty && !isImported, isImported, isBuiltIn)
+      case p: PMethodSig =>
+        GobraMemberInfo(pkgDir, pkgName, p.id.name, formatArgs(p.args), MethodSignature, isTrusted = p.spec.isTrusted, isAbstract = false, isImported, isBuiltIn)
+      case p: PFPredicateDecl =>
+        GobraMemberInfo(pkgDir, pkgName, p.id.name, formatArgs(p.args), FunctionPredicateDeclaration, isTrusted = false, p.body.isEmpty && !isImported, isImported, isBuiltIn)
+      case p: PMPredicateDecl =>
+        GobraMemberInfo(pkgDir, pkgName, p.receiver.typ.formattedShort + "." + p.id.name, formatArgs(p.args), MethodPredicateDeclaration, isTrusted = false, p.body.isEmpty && !isImported, isImported, isBuiltIn)
+      case p: PMethodImplementationProof =>
+        GobraMemberInfo(pkgDir, pkgName, "Impl_Proof." + p.receiver.typ.formattedShort + "." + p.id.name, formatArgs(p.args), MethodImplementationProof, isTrusted = false, p.body.isEmpty && !isImported, isImported, isBuiltIn)
+      case p: PDomainFunction =>
+        GobraMemberInfo(pkgDir, pkgName, p.id.name, formatArgs(p.args), DomainFunction, isTrusted = false, isAbstract = false, isImported, isBuiltIn)
+      case p: PMPredicateSig =>
+        GobraMemberInfo(pkgDir, pkgName, p.id.name, formatArgs(p.args), MethodPredicateSignature, isTrusted = false, isAbstract = false, isImported, isBuiltIn)
+      case p: PPredConstructor =>
+        GobraMemberInfo(pkgDir, pkgName, p.id.id.name, formatPredConstructorArgs(p.args), PredicateConstructor, isTrusted = false, isAbstract = false, isImported, isBuiltIn)
+      // Fallback to the node's code root if we can't match the node
+      case p: PNode => getMemberInformation(nodeTypeInfo.codeRoot(p), typeInfo)
+    }
+  }
+
+  /**
+   * Checks whether a Go Tree contains a node
+   */
+  def treeContains(tree: Info.GoTree, p: PNode): Boolean = {
+    try {
+      tree.whenContains(p, true)
+    } catch {
+      case NodeNotInTreeException(_) => false
+    }
+  }
+
+  /**
+   * Returns a set of viper member names (Methods, Predicates & Functions) which are accessed by a given viper member
+   */
+  def getViperDependencies(m: Member): Set[String] = m match {
```

Comment:
Couldn't we make use of the dependency analysis that the Chopper has already implemented instead of replicating it here?

Reply:
Good Idea!
But I didn't make use of the exact  same analysis chopper made, since it is built to isolate parts of the program.
I reused chopper's dependency method instead of writing my own